### PR TITLE
Allow editing label layouts for printers

### PR DESCRIPTION
**Pull request by apmasell**
JIRA: GLT-3123

### CODE REVIEW

File: miso-web/src/main/webapp/scripts/labeldesign.js [hunk 1]
```diff
@@ -0,0 +1,606 @@
+/*
+ * Copyright (c) 2012. The Genome Analysis Centre, Norwich, UK
+ * MISO project contacts: Robert Davey @ TGAC
```
Comment (avarsava) [inline miso-web/src/main/webapp/scripts/labeldesign.js:3]: Is this still an accurate contact for new files?

**Review by avarsava — APPROVED** [review]

File: miso-web/src/main/webapp/scripts/labeldesign.js (hunk 1 above)
Comment (apmasell) [reply miso-web/src/main/webapp/scripts/labeldesign.js:3]: Good question. Dillan, what are we doing for new files?

Comment (djcooke) [reply miso-web/src/main/webapp/scripts/labeldesign.js:3]: Mostly not including it at all

File: miso-web/src/main/webapp/scripts/list_printer.js [hunk 2]
```diff
@@ -35,15 +35,78 @@ ListTarget.printer = {
...
+          property: "width",
+          value: items[0].width
+        }, {
+          type: "float",
+          label: "Height (mm)",
+          property: "height",
+          value: items[0].height
+        }], function(settings) {
+          if (!settings.name) {
+            Utils.showOkDialog('Create Printer', ['A printer needs a name.']);
```
Comment (djcooke) [inline miso-web/src/main/webapp/scripts/list_printer.js:67]: Can set `required: true` in the field specifications above instead. For width and height too

File: miso-web/src/main/webapp/scripts/list_printer.js [hunk 3]
```diff
@@ -84,13 +143,6 @@ ListTarget.printer = {
               Utils.showOkDialog('Create Printer', ['A printer needs a name.']);
```
Comment (djcooke) [inline miso-web/src/main/webapp/scripts/list_printer.js:143]: The fields here should all be required too

File: miso-web/src/main/webapp/scripts/labeldesign.js (hunk 1 above)
Comment (apmasell) [reply miso-web/src/main/webapp/scripts/labeldesign.js:3]: I like this solution best.

**Review by djcooke — APPROVED** [review]

**Pull request merged**